go/libkb/api.go

```diff
@@ -420,14 +420,31 @@ func (a *InternalAPIEngine) getURL(arg APIArg) url.URL {
 	return u
 }
 
-func (a *InternalAPIEngine) sessionArgs(arg APIArg) (tok, csrf string) {
+func (a *InternalAPIEngine) sessionArgs(arg APIArg) (tok, csrf string, err error) {
 	if arg.SessionR != nil {
-		return arg.SessionR.APIArgs()
+		tok, csrf = arg.SessionR.APIArgs()
+		return tok, csrf, nil
 	}
-	a.G().LoginState().LocalSession(func(s *Session) {
-		tok, csrf = s.APIArgs()
+
+	a.G().LoginState().Account(func(a *Account) {
```

Comment:
Before, when the service started, sesscheck would be called if necessary and refresh your session file.  

If started offline, your session file, token could be stale.  Once online and making an API request, instead of getting a bad session error when an API request goes through, this will first refresh the session if necessary.